src/lib/storage/table.cpp

```diff
@@ -191,6 +191,40 @@ std::vector<bool> Table::columns_are_nullable() const {
   return nullable;
 }
 
+bool Table::column_is_unique(const ColumnID column_id) const {
+  Assert(column_id < _column_definitions.size(), "ColumnID out of range.");
+  for (const auto& key_constraint : soft_key_constraints()) {
+    const auto& key_type = key_constraint.key_type();
+    if (key_type != KeyConstraintType::PRIMARY_KEY && key_type != KeyConstraintType::UNIQUE) {
+      continue;
+    }
+
+    for (const auto& constraint_column_id : key_constraint.columns()) {
+      if (constraint_column_id == column_id) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+std::vector<bool> Table::columns_are_unique() const {
```

Comment:
Cannot see a usage?

If you plan to use/I missed it: the name is sounds more like sth that returns a single bool.

Reply:
Does not have a usage, I just added it for completeness. Should I remove it or rename it?

Reply:
If we don't use it, we don't need it.

Reply:
Removed.